### PR TITLE
feature: support custom optim and lr scheduler

### LINKED ISSUE #36: 支持自定义optim
cfg文件支持自定义optim，通过optim type字段类型进行选择，如果类型为str则使用内建optim；如果类型为Optimazer，表示自定义optim，不需要通过getattr直接初始化即可。

### PR DESCRIPTION
**Pull request by cnstark**
close #36 
支持在cfg文件中通过传入optim type来使用自定义optim，lr_scheduler同理。
例如：
```python
from torch_optimizer import RAdam

# ...

CFG.TRAIN.OPTIM = EasyDict()
CFG.TRAIN.OPTIM.TYPE = RAdam
CFG.TRAIN.OPTIM.PARAM = {
    'lr': 1e-3,
    'betas': [0.9, 0.999],
    'eps': 1e-8
}
```

### CODE REVIEW

File: easytorch/core/optimizer_builder.py [hunk 1]
```diff
@@ -39,22 +40,26 @@ def build_optim(optim_cfg: dict, model: nn.Module) -> optim.Optimizer:
     """
 
     optim_type = optim_cfg['TYPE']
-    if hasattr(optim, optim_type):
-        Optim = getattr(optim, optim_type)
+    if isinstance(optim_type, type):
```
Comment (zezhishao) [inline easytorch/core/optimizer_builder.py:43]: `type`改成`torch.optim.Optimizer`会不会更合适一点。

File: easytorch/core/optimizer_builder.py [hunk 2]
```diff
@@ -83,10 +88,13 @@ def build_lr_scheduler(lr_scheduler_cfg: dict, optimizer: optim.Optimizer):
     """
 
     lr_scheduler_type = lr_scheduler_cfg['TYPE']
-    if hasattr(lr_scheduler, lr_scheduler_type):
-        Scheduler = getattr(lr_scheduler, lr_scheduler_type)
+    if isinstance(lr_scheduler_type, type):
```
Comment (zezhishao) [inline easytorch/core/optimizer_builder.py:91]: 同理，这个是不是可以考虑把`type`改成`torch.optim.lr_scheduler._LRScheduler`？
或者为了兼容不按照pytorch官方的方式自定义的`lr_scheduler`，仅限定为`object`？
（假如这样做的话，上一个comment那也需要统一限定为`object`）
